schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,110 @@ title:              "Post Artifact Request"
 description: "Request a authorization to put and artifact or posting of a URL as an artifact. Note that the `storageType` property is referenced in the response as well."
 type:               object
 oneOf:
+  - title: "Blob Artifact Request"
+    description: |
+      Request a list of requests in a generalized format which can be run to
+      upload an artifact to storage managed by the queue.
+    type: object
+    properties:
+      storageType:
+        description: Artifact storage type, in this case `'blob'`
+        type: string
+        enum:
+          - blob
+      expires:
+        description: |
+          Date-time after which the artifact should be deleted. Note, that
+          these will be collected over time, and artifacts may remain
+          available after expiration. S3 based artifacts are identified in
+          azure table storage and explicitly deleted on S3 after expiration.
+        type: string
+        format: date-time
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type: string
+        maxLength: 255
+      encoding:
```

Comment:
Okay, so apart from the naming change... what do we do with `contentEncoding: gzip`, I suspect gzip might be the only content-encoding we want to support? Or do we want `br` support too? Apparently, that's spec too?

Oh, and do we want the sha256 before or after applying `content-encoding`. Reading about `content-md5` it seems that hashing should be done in canonical-form, I understand that as after `content-encoding` have been decoded... Which also makes most sense to me...

But that means we need to send two hashes, one for use as `x-amz-content-sha256` (hash with `content-encoding`) and one for use as `x-amz-meta-content-sha256` (hash without `content-encoding`). For multi-part upload this is no problem, for single part upload we would need a new key, or specify `parts[0] = ...`, and then have the top-level hash be that of the artifact, and the parts hash be those used for transfer.

Hmm, I'm not sure... We might need to pull in some advice from dustin or bstack. I'm not perfectly sure what is best here.